### DISCUSSION TITLE
Why do I have to use two mods when Courseplay done it before by itself?

### DISCUSSION THREAD

**Discussion by Ursus95** [Courseplay/Courseplay_FS22#3608] [Q&A] [answered]
I am having a lot of problems with Autodrive. In my opinion, it’s a very flawed mod, full of bugs, poor UI, and no real editing tools. I wouldn’t care except I don’t have a choice—I have to use it because CP won’t work on its own.

In previous editions, if I wanted to make a fill-and-dump course, it was smooth as butter. Setup took 2 minutes, and then it worked flawlessly. If I wanted to combine a field course with a road course, it only took a click. I could automate my whole farm, and it took no time while working without fail.

Now I have to use two mods, the setup takes much more time, and it only works about 30% of the time. In one game session, I used the exact same course with the exact same machinery. The first time, it worked okay; the second time, it refused to work and froze my game.

Please help me understand why the road course system had to be removed. It was simple, solid, and intuitive.

**Comment by Tensuko** [comment, marked as answer]
What ppl don't understand is that we did not remove it, we did not re-implement it.
Reason is that we have much more to support and to maintain with that mode, wich all adds up with triggers, UI and all that stuff and we decided to leave that to AutoDrive. We are also not that happy how AutoDrive turned out, we expected the cooperation a bit different, but many Users are happy as how it works. So we will keep it like that for now.
The old simple A-B Routes won't be added back.
I will not talk about the Plans we have, as we can't promise anything at anytime, but we have things in mind to implement.
But either solutions, will take time to implement.

And to be honest, for quick and dirty solutions, AutoDrive works...

**Reply by rohne83** [reply]
For delivery/reload at silos far off the fields, AD does quite a good job and even better than CP did. That's where AD has its strengths. And I never had any issues with AD when it comes to that. Most of the time issues with AD on the road are a result to bad settings.
It's rather the refuelling of sowing machines, slurry tankers etc at field edge which is quite uncomfortable with AD.

**Reply by Tensuko** [reply]
> It's rather the refuelling of sowing machines, slurry tankers etc at field edge which is quite uncomfortable with AD.

That we have on todo :)

**Reply by coatsy35** [reply]
What annoys me is when using AD to fill a lorry from a chaser, if the chaser is larger than the lorry, the chaser just sits there and doesn’t resume to the combine, when the lorry goes.

**Reply by Ursus95** [reply]
Thank you for the answer.

I understand that maintaining a mod as detailed as this with such a small team is very challenging. However, you can understand the frustration when important features are discontinued and cannot be replaced with other mods.
After spending many hours learning and watching tutorials for AD, the mod still falls far short of being usable. It is very finicky, bugs out over the smallest things, lacks editing tools, and consumes a lot of resources. I tried using it for the simplest tasks and still failes.

That said, I also want to give the team credit for the significant progress made in other parts of the mod. For example, the combine unload feature is leaps and bounds ahead in terms of the AI's cleverness compared to previous versions of the mod.
